agent/stats/engine.go

```diff
@@ -228,59 +257,109 @@ func (engine *DockerStatsEngine) removeAll() {
 	}
 }
 
+func (engine *DockerStatsEngine) getStatsTaskContainer(task *apitask.Task, dockerContainerName string) {
```

Comment:
to align with addToStatsContainerMapUnsafe, addToStatsContainerMapUnsafe is probably a better name

Reply:
also, i think we should just inspect the pause container and use its pid instead of arbitrary one, since pause container is guaranteed to be running when task is running while other container can be non-essential and stop before the task stops